src/creeps/Zombie.cpp

```diff
@@ -117,41 +124,82 @@ void Zombie::generateMove() {
+            setSrcRect(getSrcRect().x, ZOMBIE_FRONT_RIGHT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_FRONT_RIGHT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            }
             break;
         case ZombieDirection::DIR_D:
             setDX(0);
             setDY(ZOMBIE_VELOCITY);
             setAngle(static_cast<double>(ZombieAngles::SOUTH));
+            setSrcRect(getSrcRect().x, ZOMBIE_FRONT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_FRONT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            }
             break;
         case ZombieDirection::DIR_LD:
             setDX(-ZOMBIE_VELOCITY);
             setDY(ZOMBIE_VELOCITY);
             setAngle(static_cast<double>(ZombieAngles::SOUTHWEST));
+            setSrcRect(getSrcRect().x, ZOMBIE_FRONT_LEFT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_FRONT_LEFT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            }
             break;
         case ZombieDirection::DIR_L:
             setDX(-ZOMBIE_VELOCITY);
             setDY(0);
             setAngle(static_cast<double>(ZombieAngles::WEST));
+            setSrcRect(getSrcRect().x, ZOMBIE_LEFT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_LEFT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            }
             break;
         case ZombieDirection::DIR_LU:
             setDX(-ZOMBIE_VELOCITY);
             setDY(-ZOMBIE_VELOCITY);
             setAngle(static_cast<double>(ZombieAngles::NORTHWEST));
+            setSrcRect(getSrcRect().x, ZOMBIE_BACK_LEFT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_BACK_LEFT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            }
             break;
         case ZombieDirection::DIR_U:
             setDX(0);
             setDY(-ZOMBIE_VELOCITY);
             setAngle(static_cast<double>(ZombieAngles::NORTH));
+            setSrcRect(getSrcRect().x, ZOMBIE_BACK, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_BACK, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            }
             break;
         case ZombieDirection::DIR_RU:
             setDX(ZOMBIE_VELOCITY);
             setDY(-ZOMBIE_VELOCITY);
             setAngle(static_cast<double>(ZombieAngles::NORTHEAST));
+            setSrcRect(getSrcRect().x, ZOMBIE_BACK_RIGHT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
+            updateZombieWalk(direction);
+            if(pathLength == stepsLeft) {
+                setSrcRect(0, ZOMBIE_BACK_RIGHT, ZOMBIE_WIDTH, ZOMBIE_HEIGHT);
```

Comment:
All of the added changes are the same across all directions except an invalid direction
So just do an early return in the invalid case, and move this new section down below the switch statement
That way we reduce a lot of code duplication
There's no reason to have 8 copies of the same code